contrib/platform/src/com/sun/jna/platform/mac/CoreFoundation.java

```diff
@@ -0,0 +1,1029 @@
+         * this method returns an approximate value.
+         *
+         * @return The corresponding {@code float}
+         */
+        public float floatValue() {
+            FloatByReference fbr = new FloatByReference();
+            INSTANCE.CFNumberGetValue(this, CFNumberType.kCFNumberFloatType.typeIndex(), fbr);
+            return fbr.getValue();
+        }
+    }
+
+    /**
+     * Enum of values used for {@link CFNumberType} in {@link #CFNumberGetValue} and
+     * {@link #CFNumberGetType}. Use {@link CFNumberType#typeIndex} for the expected
+     * integer value corresponding to the C-style enum.
+     */
+    enum CFNumberType {
+        unusedZero, kCFNumberSInt8Type, kCFNumberSInt16Type, kCFNumberSInt32Type, kCFNumberSInt64Type,
+        kCFNumberFloat32Type, kCFNumberFloat64Type, kCFNumberCharType, kCFNumberShortType, kCFNumberIntType,
+        kCFNumberLongType, kCFNumberLongLongType, kCFNumberFloatType, kCFNumberDoubleType, kCFNumberCFIndexType,
+        kCFNumberNSIntegerType, kCFNumberCGFloatType, kCFNumberMaxType;
+
+        /**
+         * Index for the type of {@link CFNumberRef} stored.
+         *
+         * @return a {@link CFIndex} representing the enum ordinal.
+         */
+        public CFIndex typeIndex() {
+            return new CFIndex(this.ordinal());
+        }
+    }
+
+    /**
+     * A reference to a {@code CFBoolean} object.
+     */
+    class CFBooleanRef extends CFTypeRef {
+        public CFBooleanRef() {
+            super();
+        }
+
+        public CFBooleanRef(Pointer p) {
+            super(p);
+            if (!isTypeID(BOOLEAN_TYPE_ID)) {
+                throw new ClassCastException("Unable to cast to CFBoolean. Type ID: " + getTypeID());
+            }
+        }
+
+        /**
+         * Convert a reference to a Core Foundations Boolean into its {@code boolean}
+         *
+         * @return The corresponding {@code boolean}
+         */
+        public boolean booleanValue() {
+            return 0 != INSTANCE.CFBooleanGetValue(this);
+        }
+    }
+
+    /**
+     * A reference to an immutable {@code CFArray} object.
+     * <p>
+     * CFArray is “toll-free bridged” with its Cocoa Foundation counterpart,
+     * {@code NSArray}. Therefore, in a method where you see an {@code NSArray *}
+     * parameter, you can pass in a {@link #CFArrayRef}.
+     */
+    class CFArrayRef extends CFTypeRef {
+        public CFArrayRef() {
+            super();
+        }
+
+        public CFArrayRef(Pointer p) {
+            super(p);
+            if (!isTypeID(ARRAY_TYPE_ID)) {
+                throw new ClassCastException("Unable to cast to CFArray. Type ID: " + getTypeID());
+            }
+        }
+
+        /**
+         * Convenience method for {@link #CFArrayGetCount} on this object
+         *
+         * @return The number of values in this array.
+         */
+        public CFIndex getCount() {
```

Comment:
Would it make sense to return an int/long here?

Reply:
Yes, it would.  My use case for this (which is in one of the test classes) is reading the mac power source status, and as currently implemented requires:
```
int powerSourcesCount = powerSourcesList.getCount().intValue();
for (int ps = 0; ps < powerSourcesCount; ps++) {
    Pointer pwrSrcPtr = powerSourcesList.getValueAtIndex(new CFIndex(ps));
    ...
}
```
But as you say it would be so much simpler to do:
```
int powerSourcesCount = powerSourcesList.getCount();
for (int ps = 0; ps < powerSourcesCount; ps++) {
    Pointer pwrSrcPtr = powerSourcesList.getValueAtIndex(ps);
    ...
}
```
Also, `int` will probably always be preferable to `long` as we're dealing with arrays, which require an `int` index in Java.